app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1013,31 +1013,34 @@ public void onClick(DialogInterface dialog, int i) {
         }
     }
 
+    /**
+     * Create (or re-use) a form record and pass it to the form entry activity
+     * launcher. If there is an existing incomplete form that uses the same
+     * case, ask the user if they want to edit or delete that one.
+     *
+     * @param state Needed for FormRecord manipulations
+     */
     private void startFormEntry(AndroidSessionWrapper state) throws SessionUnavailableException {
-        try {
-            //If this is a new record (never saved before), which currently all should be 
             if (state.getFormRecordId() == -1) {
-
-                //If form management isn't enabled we can't have these old forms around anyway
-                if (CommCarePreferences.isIncompleteFormsEnabled()) {
-                    //First, see if we've already started this form before
+                if (CommCarePreferences.isIncompleteFormsEnabled() &&
+                        state.getSessionStateDescriptor().getSessionDescriptor().contains(SessionFrame.STATE_DATUM_VAL)) {
```

Comment:
searchForDuplicates checks for SessionStateDescriptors that match the current one, so if the current one doesn't contain 'STATE_DATUM_VAL', then why even bother running searchForDuplicates, right?

Based on this assumption I've re-ordered the code to only run searchForDuplicates if the current SessionStateDescriptor contains 'STATE_DATUM_VAL'